packages/vite/src/node/server/index.ts

```diff
@@ -779,6 +779,11 @@ export async function _createServer(
   return server
 }
 
+// When using non strict port for the dev server, the running port can be different from the config one.
+// When restarting, the original port may be available but to avoid a switch of URL for the running
+// browser tabs, we enforce the previously used port, expect if the config port changed.
+let _currentServerPort: number | undefined
+let _configServerPort: number | undefined
```

Comment:
These global variables will interfere with creating multiple dev server in the same process.

Let's move them as internal properties of the server, and we can forward them to the new server when restarting here before the `Object.assign`: https://github.com/vitejs/vite/blob/b5653d3075559a327b8d87a3863be3260d880e3a/packages/vite/src/node/server/index.ts#L931

Reply:
Okay, I've moved them.

Reply:
Thanks!